### PR TITLE
test(inspect): add coverage for pure-function edge cases and text-positioning operators

### PR DESCRIPTION
**Pull request by mitsuru**
# Pull Request

## Summary

`inspect.rs` の行カバレッジが 91.28% と低かったため、未カバー分岐にテストを追加しました。
今回はテストのみの変更で、プロダクションコードへの変更はありません。

**カバレッジ推移（行カバレッジ）:**

| ファイル | 変更前 | 変更後 | 差分 |
|---------|--------|--------|------|
| `inspect.rs` | 91.28% (47 missed) | 96.10% (27 missed) | **+4.82pp** |

## Changes

- `obj_to_f32_returns_zero_for_non_numeric_object`: `obj_to_f32` の `_ => 0.0` フォールバック arm をカバー（Null / Boolean / Name を渡すケース）
- `detect_image_format_unrecognized_name_filter`: Filter が認識されない Name（"ASCII85Decode" 等）のとき "unknown" を返す `_ => {}` arm をカバー
- `detect_image_format_non_name_non_array_filter_object`: Filter が Name でも Array でもない Object（Integer 等）のとき `_ => String::new()` arm をカバー
- `inspect_metadata_returns_all_none_when_no_info_dict`: trailer に Info キーがない PDF で `Err(_) => return meta` ブランチをカバー
- `make_pdf_with_text_positioning_ops` + 3テスト: lopdf で最小 PDF を直接構築し、fulgur 生成 PDF では使われない `TL` / `Td` / `TD` / `T*` テキスト配置オペレーターをカバー（fulgur は Tm/TJ を使うため、これらは合成 PDF でのみ到達可能なコードパス）

**意図的に残した未カバー箇所（理由）:**

- `extract_metadata` 78行目 (`Info` が参照でなくインライン値の場合)・84行目 (`Info` が Dictionary でない場合): lopdf の通常の API でこれらの状態を作るのは非現実的（lopdf 内部の不変条件に反する）
- `extract_text_items` 109・113行目 (page content ロード/デコードエラー): 壊れた PDF バイナリを手動で作成する必要があり、テストの保守コストに対してリターンが小さい
- `resolve_page_resources` の親チェーンたどり (291–295行目): fulgur 生成 PDF はページに Resources を直接持つため、親チェーンが必要な構造を lopdf で再現するには追加の PDF 構造を組み立てる必要があり、今回のスコープ外とした

## Test plan

- [x] `cargo test -p fulgur --lib`（1624 passed）
- [x] `cargo clippy -- -D warnings`（警告なし）
- [x] `cargo fmt --check`（差分なし）
- [ ] `npx markdownlint-cli2 '**/*.md'`（Markdown 変更なし）

## Contributor License Agreement

By opening this pull request, I confirm that:

- [x] I have read the [Contributing Guide](https://github.com/fulgur-rs/fulgur/blob/main/CONTRIBUTING.md)
- [x] I have read the [CLA](https://github.com/fulgur-rs/fulgur/blob/main/CLA.md)
      and agree to its terms. (First-time contributors: the CLA Assistant bot
      will comment on this PR with sign-in instructions.)

## Related issues

カバレッジ自動改善ルーティン（2026-07-05）による定期実行。

---
_Generated by [Claude Code](https://claude.ai/code/session_0122KLrUkoM31S1hUoJmeaR6)_

## Summary by CodeRabbit

* **Bug Fixes**
  * PDFのテキスト抽出で、文字位置の変化をより正しく扱えるようになりました。
  * 一部の画像判定や未対応形式の検出が安定し、未知の入力でも安全に「不明」と扱うようになりました。
  * メタデータがないPDFでも、情報の欠落を適切に処理するようになりました。

### CODE REVIEW

File: crates/fulgur/src/inspect.rs [hunk 1]
```diff
@@ -728,4 +728,245 @@ mod tests {
...
+            }),
+        );
+        let catalog_id = doc.add_object(dictionary! {
+            "Type" => "Catalog",
+            "Pages" => pages_id,
+        });
+        doc.trailer.set("Root", Object::Reference(catalog_id));
+
+        let mut buf = Vec::new();
+        doc.save_to(&mut std::io::Cursor::new(&mut buf)).unwrap();
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/inspect.rs]: ご指摘ありがとうございます。`Vec<u8>` が `std::io::Write` を直接実装しているため、`Cursor` ラッパーは不要でした。`b4aa781` で修正済みです。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/inspect.rs [hunk 2]
```diff
@@ -728,4 +728,245 @@ mod tests {
...
+        // After `Td 100 700`, tx should be ~100 + advance; verify the first
+        // item's x is near 100 (within the default identity CTM).
+        let pdf = make_pdf_with_text_positioning_ops();
+        let result = inspect_bytes(&pdf);
+        let first = result.text_items.first().expect("expected text items");
+        assert!(
+            first.x > 50.0,
+            "text x after Td 100 should be > 50, got {}",
+            first.x
+        );
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/inspect.rs:869]: ご指摘の通りです。恒等 CTM 下では `Td 100 700` 後の x 座標は正確に `100.0` になるため、`(first.x - 100.0).abs() < 1e-4` で厳密に検証するよう `b4aa781` で修正しました。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/inspect.rs [hunk 3]
```diff
@@ -728,4 +728,245 @@ mod tests {
...
+        assert!(result.text_items.len() >= 3, "need at least 3 text items");
+        let y1 = result.text_items[1].y;
+        let y2 = result.text_items[2].y;
+        // y2 (after T*) should differ from y1 (after TD) by ~14 pt in either
+        // direction (depending on the coordinate convention in use).
+        let diff = (y2 - y1).abs();
+        assert!(
+            diff > 1.0,
+            "T* should advance y from TD-positioned line; diff was {diff}"
+        );
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/inspect.rs:888]: ご指摘ありがとうございます。`TD 0 -14` で `text_leading = 14.0` がセットされ、`T*` で y が正確に `14.0` 進むはずなので、`(diff - 14.0).abs() < 1e-4` に変更して厳密に検証するよう `b4aa781` で修正しました。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/inspect.rs [hunk 4]
```diff
@@ -728,4 +728,245 @@ mod tests {
...
+        let catalog_id = doc.add_object(dictionary! {
+            "Type" => "Catalog",
+            "Pages" => pages_id,
+        });
+        // No trailer "Info" key → exercises the Err branch in extract_metadata
+        doc.trailer.set("Root", Object::Reference(catalog_id));
+
+        let mut buf = Vec::new();
+        doc.save_to(&mut std::io::Cursor::new(&mut buf)).unwrap();
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/inspect.rs]: こちらも同様に `Cursor` ラッパーを削除し、`doc.save_to(&mut buf)` を直接呼ぶよう `b4aa781` で修正済みです。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/inspect.rs [hunk 5]
```diff
@@ -728,4 +728,245 @@ mod tests {
...
+    #[test]
+    fn obj_to_f32_returns_zero_for_non_numeric_object() {
+        // The `_ => 0.0` arm is hit when the Object is neither Integer nor Real
+        // (e.g., a Name or Null). This covers the fallback branch.
+        assert_eq!(obj_to_f32(&lopdf::Object::Null), 0.0);
+        assert_eq!(obj_to_f32(&lopdf::Object::Boolean(true)), 0.0);
+        assert_eq!(obj_to_f32(&lopdf::Object::Name(b"F1".to_vec())), 0.0);
+        // The covered variants
+        assert_eq!(obj_to_f32(&lopdf::Object::Integer(5)), 5.0);
+        assert!((obj_to_f32(&lopdf::Object::Real(3.14)) - 3.14).abs() < 1e-4);
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/inspect.rs]: ご指摘ありがとうございます。`3.14` が `clippy::approx_constant` を引き起こすことは同じ問題を lintジョブのログから確認しており、`b4aa781` で `2.5` に修正済みです。

---
_Generated by [Claude Code](https://claude.ai/code)_
